pdns/validate.cc

```diff
@@ -414,35 +405,41 @@ vState getKeysFor(DNSRecordOracle& dro, const DNSName& zone, keyset_t &keyset)
       LOG("ended up with zero valid DNSKEYs, going Bogus"<<endl);
       return Bogus;
     }
-    LOG("situation: we have one or more valid DNSKEYs for ["<<*zoneCutIter<<"] (want ["<<zone<<"])"<<endl);
+    LOG("situation: we have one or more valid DNSKEYs for ["<<current_cut<<"] (want ["<<zone<<"])"<<endl);
 
-    if(zoneCutIter == zoneCuts.cend()-1) {
+    if(current_cut == zone) {
       LOG("requested keyset found! returning Secure for the keyset"<<endl);
       keyset.insert(validkeys.cbegin(), validkeys.cend());
       return Secure;
     }
 
     // We now have the DNSKEYs, use them to validate the DS records at the next zonecut
-    LOG("next name ["<<*(zoneCutIter+1)<<"], trying to get DS"<<endl);
+    DNSName next_cut = getNextCut(current_cut, zone, dro);
+    if (next_cut == current_cut) {
```

Comment:
I'm not sure in what case this might happen? It seems to me that `getNextCut()` will never return `current_cut`, except if `current_cut` is equal to `zone` which is handled a few lines above. Not an issue, of course, but I'm wondering if I'm missing something here.

Reply:
This is a 'just in case' so we always break out of the loop. `getNextCut` returns `current_cut` if it cannot find one.

Reply:
It looks like it would return `zone`, not `current_cut` (ie `target` and not `current` inside `getNextCut()`)?

Reply:
looks like you're right, fixing